s4e-backend/src/main/java/pl/cyfronet/s4e/data/repository/SceneRepositoryImpl.java

```diff
@@ -0,0 +1,154 @@
+                resultQuery.append(params.get("sensingTo"));
+                resultQuery.append("' ");
+            }
+        }
+    }
+
+    private void queryByIngestionTime(Map<String, Object> params, StringBuilder resultQuery) {
+        ZoneParameter tz = new ZoneParameter(ZoneId.of((String) params.getOrDefault("tz", "UTC")));
+        if (params.containsKey("ingestionFrom") && params.containsKey("ingestionTo")) {
+            // [TIMESTAMP] ingestionFrom / ingestionTo -> ingestion_time
+            resultQuery.append(" AND to_timestamp(s.metadata_content->>'ingestion_time', 'YYYY-MM-DD\"T\"HH24:MI:SS\"Z\"')");
+            resultQuery.append("BETWEEN '" + params.get("ingestionFrom"));
+            resultQuery.append("' AND '" + params.get("ingestionTo"));
+            resultQuery.append("' ");
+        } else {
+            if (params.containsKey("ingestionFrom")) {
+                resultQuery.append(" AND to_timestamp(s.metadata_content->>'ingestion_time', 'YYYY-MM-DD\"T\"HH24:MI:SS\"Z\"') >= '");
+                resultQuery.append(params.get("ingestionFrom"));
+                resultQuery.append("' ");
+            }
+            if (params.containsKey("ingestionTo")) {
+                resultQuery.append(" AND to_timestamp(s.metadata_content->>'ingestion_time', 'YYYY-MM-DD\"T\"HH24:MI:SS\"Z\"') >= '");
+                resultQuery.append(params.get("ingestionTo"));
+                resultQuery.append("' ");
+            }
+        }
+    }
+
+    private void queryBySatellitePlatform(Map<String, Object> params, StringBuilder resultQuery) {
+        if (params.containsKey("satellitePlatform")) {
+            // [string] satellitePlatform -> spacecraft
+            resultQuery.append(" AND s.metadata_content->>'spacecraft' = '");
+            resultQuery.append(params.get("satellitePlatform"));
+            resultQuery.append("' ");
+        }
+    }
+
+    private void queryByProductType(Map<String, Object> params, StringBuilder resultQuery) {
+        if (params.containsKey("productType")) {
+            // [string] productType -> product_type
+            resultQuery.append(" AND s.metadata_content->>'product_type' = '");
+            resultQuery.append(params.get("productType"));
+            resultQuery.append("' ");
+        }
+    }
+
+    private void queryByPolarisation(Map<String, Object> params, StringBuilder resultQuery) {
+        if (params.containsKey("polarisation")) {
+            // [string] polarisation -> polarisation
+            resultQuery.append(" AND s.metadata_content->>'polarisation' = '");
+            resultQuery.append(params.get("polarisation"));
+            resultQuery.append("' ");
+        }
+    }
+
+    private void queryByCloudCover(Map<String, Object> params, StringBuilder resultQuery) {
+        if (params.containsKey("cloudCover")) {
+            // [FLOAT] cloudCover -> cloud_cover
+            resultQuery.append(" AND (s.metadata_content->>'cloud_cover')::float = '");
```

Comment:
Querying by cloud cover should allow finding scenes which have it below the threshold.

Reply:
done